fusequery/query/src/pipelines/transforms/transform_expression_executor.rs

```diff
@@ -51,7 +52,11 @@ impl ExpressionExecutor {
         Ok(())
     }
 
-    pub fn execute(&self, block: &DataBlock) -> Result<DataBlock> {
+    pub fn execute(
+        &self,
+        block: &DataBlock,
+        exists_res: Option<&HashMap<String, bool>>,
```

Comment:
Introduce `HashMap<String, bool> ` is too limited just for exists sub query,  I think we should introduce a new struct to store the subquery results. Such as 

```
struct SubqueryResult {
       blocks: Map<String, DataBlock>
}
```

Reply:
I see, this is much cleaner. I just tried this approach and it turns out `DataBlock` is not serializable or deserializable.
To fix this, I need to add  `#[derive(serde::Serialize, serde::Deserialize)]` to `DataBlock` and `DataColumnarValue` and `DataArrayRef`. 

However, `DataArrayRef` is imported from arrow. I am not sure how to add the de/serialization trait for them.
Is there any workaround?